public/js/reader.js

```diff
@@ -315,16 +316,44 @@ Reader.handleShortcuts = function (e) {
     case 27: // escape
         LRR.closeOverlay();
         break;
-    case 32: // spacebar
-        if ($(".page-overlay").is(":visible")) { break; }
-        if (e.originalEvent.getModifierState("Shift") && (window.scrollY) === 0) {
-            (Reader.mangaMode) ? Reader.changePage(1) : Reader.changePage(-1);
-        } else if (($(window).height() + $(window).scrollTop()) >= LRR.getDocHeight()) {
-            (Reader.mangaMode) ? Reader.changePage(-1) : Reader.changePage(1);
-        }
-        // spacebar is always forward regardless of reading direction, so it needs to be flipped
-        // to always result in a positive offset when it reaches the changePage() logic
+    case 32: { // spacebar
+    //Break if overlay is open, browser detects repeatkey but not held, or webtoon gallery and in infinite scroll
+    if ($(".page-overlay").is(":visible") || e.repeat || (Reader.infiniteScroll && Reader.tags?.includes("other:webtoon"))) break;
+    e.preventDefault();
```

Comment:
Since this function is only fired on `keyup`, I still get the browser default behavior when I press space.  

This means that on default settings:  
- When holding down space, the reader scrolls down to the bottom of the current page
- When releasing the key, this function triggers and it goes to the next page

I don't actually hate this -- It's one less keystroke to fully read a page and switch to the next!  
I've found it to be inconsistent when quickly pressing and releasing space however, it sometimes does both steps and scrolls + goes to next page, or only scrolls down.  

Should we just add another `preventDefault` on `keydown` for spacebar to kill the default browser behavior?

Reply:
I tried preventDefault first, but it caused the function to just not work at all. If I don't use either function then pushing the spacebar for 0.2 seconds could result in it activating twice, it was so sensitive that it became difficult to push and let go of space fast enough for it to only register once.

I'll give it another shot and see if I can get it to work on keydown instead.

Reply:
Tried your suggestion to disable keydown but after spending some hours testing.. I don't like it not being able to scroll when held with normal browser function. But allowing browser scroll on key.repeat then blocking on keydown doesnt work because it will skip the repeat check in first instance due to not being a repeat on first run, then it just breaks and will never hit the key.repeat as its already been pushed and hit the break.

Not sure how I can handle this

Reply:
hmm, well I don't think I have any better ideas.  
As it stands this is already better behavior and the inconsistency really only happens if you're used to the old way it works... We'll see how it goes with users, imo fixing the snap issue is a bigger win.